river/Decoration.zig

```diff
@@ -49,8 +56,24 @@ fn handleDestroy(listener: ?*c.wl_listener, data: ?*c_void) callconv(.C) void {
 
 fn handleRequestMode(listener: ?*c.wl_listener, data: ?*c_void) callconv(.C) void {
     const self = @fieldParentPtr(Self, "listen_request_mode", listener.?);
-    _ = c.wlr_xdg_toplevel_decoration_v1_set_mode(
-        self.wlr_xdg_toplevel_decoration,
-        .WLR_XDG_TOPLEVEL_DECORATION_V1_MODE_SERVER_SIDE,
-    );
+
+    const wlr_xdg_surface: *c.wlr_xdg_surface = self.wlr_xdg_toplevel_decoration.surface;
+    const wlr_xdg_toplevel: *c.wlr_xdg_toplevel = @field(wlr_xdg_surface, c.wlr_xdg_surface_union).toplevel;
+    const app_id: [*:0]const u8 = if (wlr_xdg_toplevel.app_id) |id| id else "NULL";
+
+    const use_csd = for (self.server.config.csd_filter.items) |filter_app_id| {
+        if (std.mem.eql(u8, std.mem.span(app_id), std.mem.span(filter_app_id))) break true;
```

Comment:
Don't need the span here as it's already a slice
```suggestion
        if (std.mem.eql(u8, std.mem.span(app_id), filter_app_id)) break true;
```